metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1294,8 +1315,11 @@ void handleBrokerFenced(int brokerId, List<ApiMessageAndVersion> records) {
      */
     void handleBrokerUnregistered(int brokerId, long brokerEpoch,
                                   List<ApiMessageAndVersion> records) {
-        generateLeaderAndIsrUpdates("handleBrokerUnregistered", brokerId, NO_LEADER, records,
-            brokersToIsrs.partitionsWithBrokerInIsr(brokerId));
+        generateLeaderAndIsrUpdates("handleBrokerUnregistered", brokerId, NO_LEADER, NO_LEADER, records,
+            new PartitionsOnReplicaIteratorChain(Arrays.asList(
```

Comment:
We can avoid using the iterator chain by calling generateLeaderAndIsrUpdates twice. I am open to both ways.

Reply:
If it's not needed for correctness, yes lets remove the chained iterator, it's a bit confusing.

Reply:
Removed.